languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/model/JavaModelInterfaceRenderer.kt

```diff
@@ -91,58 +103,46 @@ class JavaModelInterfaceRenderer @Inject constructor(override val vrapTypeProvid
 
     private fun ObjectType.staticOfMethod() : String {
         val vrapType = vrapTypeProvider.doSwitch(this) as VrapObjectType
-
-        val constructorArguments : String = this.allProperties
-                .filter { it.name != this.discriminator() }
-                .map { if(it.isPatternProperty()) "final Map<String, ${it.type.toVrapType().simpleName()}> values"
-                else if(!it.required) "@Nullable final ${it.type.toVrapType().simpleName()} ${it.name}"
-                else "final ${it.type.toVrapType().simpleName()} ${it.name}"}
-                .joinToString(separator = ", ")
-
-        val constructorValues : String = this.allProperties
-                .filter { it.name != this.discriminator() }
-                .map { if(it.isPatternProperty()) "values" else it.name }
-                .joinToString(separator = ", ")
-
         return if(this.isAbstract()) {
             ""
         }else {
             """
-                |public static ${vrapType.simpleClassName}Impl of(${constructorArguments.escapeAll()}){
-                |   return new ${vrapType.simpleClassName}Impl($constructorValues);
+                |public static ${vrapType.simpleClassName}Impl of(){
+                |   return new ${vrapType.simpleClassName}Impl();
                 |}
                 |
              """.trimMargin()
         }
     }
     
-    /**
-     * This method returns a list of required properties for the specified {@param rootObjectType} and for it's parent types.
-     */
-    private fun allRequiredProperties(rootObjectType : ObjectType) : List<Property> {
-        var currentType: ObjectType? = rootObjectType
-        val properties : MutableList<Property> = mutableListOf()
-        while (currentType != null) {
-            val currentTypeDiscriminator : String = currentType.discriminator() ?: ""
-            properties.addAll(currentType.properties.filter { it.name != currentTypeDiscriminator && it.required && !it.isPatternProperty()})
-            currentType = if(currentType.type is ObjectType) {
-                currentType.type as ObjectType
-            }else{
-                null
-            }
+    private fun ObjectType.staticOfJsonMethod() : String {
+        val vrapType = vrapTypeProvider.doSwitch(this) as VrapObjectType
+        return if(this.isAbstract()) {
+            ""
+        }else {
+            """
+                |public static ${vrapType.simpleClassName}Impl of(String json) {
```

Comment:
Hmm, do you plan to keep this? Or is it just a temporary helper method?

Reply:
I plan to keep it.